internal/test/wmcb/wmcb_test.go

```diff
@@ -70,28 +68,36 @@ var (
 		sha:     "a88e7a1c6f72ea6073dbb4ddfe2e7c8bd37c9a56d94a33823f531e303a9915e7a844ac5880097724e44dfa7f4a9659d14b79cc46e2067f6b13e6df3f3f1b0f64",
 		shaType: "sha512",
 	}
+	// cniPluginPkgName is the user-defined name of the required cni plugins package package
+	CniPluginPkgName = pkgName("cniPlugins")
 )
 
 // wmcbVM is a wrapper for the WindowsVM interface that associates it with WMCB specific testing
 type wmcbVM struct {
 	e2ef.TestWindowsVM
 }
 
-// pkg encapsulates information about a package
-type pkgInfo struct {
-	// url is the URL of the package
-	url string
-	// sha is the SHA hash of the package
-	sha string
-	// shaType is the type of SHA used, example: 256 or 512
-	shaType string
-}
-
 type wmcbFramework struct {
 	// TestFramework holds the instantiation of test suite being executed
 	*e2ef.TestFramework
-	// cniPlugins contains information about the CNI plugin package
-	cniPlugins pkgInfo
+	// contains map of packages to be downloaded
+	pkgs map[pkgName]pkgInfo
+}
+
+// initializePackages sets up all the required packages of type pkgInfo
+func (f *wmcbFramework) initializePackages() error {
+	var pkgs = make(map[pkgName]pkgInfo)
+	// create pkgInfo struct for cni plugins and populate it
+	cniPluginsPackage, err := initializePkg(CniPluginPkgName, cniPluginsBaseURL, framework.LatestCniPluginsVersion)
+	if err != nil {
+		return err
+	}
+	cniPkgInfo := cniPluginsPackage.(*cniPlugins).pkgInfo
```

Comment:
This variable is not needed. Moreover if you had used the interface in the `pkgs` map, you would not have needed to do this sort of casting. I suggest you go that route.